reproject/healpix/high_level.py

```diff
@@ -0,0 +1,90 @@
+from .core import healpix_to_image, image_to_healpix
+from .utils import parse_input_healpix_data, parse_coord_system
+from ..utils import parse_input_data, parse_output_projection
+
+__all__ = ['reproject_from_healpix', 'reproject_to_healpix']
+
+
+def reproject_from_healpix(input_data, output_projection, shape_out=None, hdu_in=None):
+    """
+    Reproject data from a HEALPIX projection to a standard projection.
+
+    Parameters
+    ----------
+    input_data : str or `~astropy.io.fits.TableHDU` or `~astropy.io.fits.BinTableHDU` or tuple
+        The input data to reproject. This can be:
+
+            * The name of a HEALPIX FITS file
+            * A `~astropy.io.fits.TableHDU` or `~astropy.io.fits.BinTableHDU`
+              instance
+            * A tuple where the first element is a `~numpy.ndarray` and the
+              second element is a `~astropy.coordinates.BaseCoordinateFrame`
+              instance or a string alias for a coordinate frame.
+
+    output_projection : `~astropy.wcs.WCS` or `~astropy.io.fits.Header`
+        The output projection, which can be either a `~astropy.wcs.WCS`
+        or a `~astropy.io.fits.Header` instance.
+    shape_out : tuple, optional
+        If ``output_projection`` is a `~astropy.wcs.WCS` instance, the
+        shape of the output data should be specified separately.
+    hdu_in : int or str, optional
+        If ``input_data`` is a FITS file, specifies the HDU to use.
+
+    Returns
+    -------
+    array_new : `~numpy.ndarray`
+        The reprojected array
+    footprint : `~numpy.ndarray`
+        Footprint of the input array in the output array. Values of 0 indicate
+        no coverage or valid values in the input image, while values of 1
+        indicate valid values. Intermediate values indicate partial coverage.
```

Comment:
Good catch, and I agree, a more detailed description of footprints would be good.